### DISCUSSION TITLE
[RFC] Release strategy

### DISCUSSION THREAD

**Discussion by dcyoung-dev** [DCC-EX/EX-WebThrottle2#31] [Ideas]
### Problem

V2 has not been released for a while. We have a `next-release` branch that is behind the `main` branch. Although fast forwarding the branch to be inline with `main` it seems like an unnecessary step.

### Alternative

Setup CI to deploy to GitHub pages when a new [GitHub Release](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository) is created.

#### Pros

- When used with semantic versioning, these releases can produce changelogs based on PRs that have been merged since the last release.
- The CI workflow for this is straight forward - running the deploy on new release - less branches == less management.
- By using this approach we could be releasing/deploying often, which lets us show what has been built quickly
- I think GitHub creates an artefact on release allowing tinkers to download the sourcecode for each release (would need to check that this happens by default)

**Comment by dexslab** [comment]
Yes we can use CI to deploy to static hosting. The goal of this was to allow easy updates and what not. Next release branch was made to facilitate collaborated work for testing before it was pushed to the main branch. Next-release is behind because i pushed to the main for the changes i made to CI
